src/JSPropUpdater.ts

```diff
@@ -55,11 +66,49 @@ export class JSPropUpdater {
     animatedComponent: React.Component<unknown, unknown>
   ) {
     const viewTag = findNodeHandle(animatedComponent);
-    JSPropUpdater._tagToComponentMapping.delete(viewTag);
-    if (JSPropUpdater._tagToComponentMapping.size === 0) {
+    JSPropUpdaterPaper._tagToComponentMapping.delete(viewTag);
+    if (JSPropUpdaterPaper._tagToComponentMapping.size === 0) {
       this._reanimatedEventEmitter.removeAllListeners(
         'onReanimatedPropsChange'
       );
     }
   }
 }
+
+class JSPropUpdaterFabric implements JSPropUpdater {
+  private static _tagToComponentMapping = new Map();
+  private static isInitialized = false;
+
+  constructor() {
+    if (!JSPropUpdaterFabric.isInitialized) {
+      const updater = (viewTag: number, props: unknown) => {
+        const component =
+          JSPropUpdaterFabric._tagToComponentMapping.get(viewTag);
+        component?._updateFromNative(props);
+      };
+      runOnUIImmediately(() => {
+        'worklet';
+        global._jsPropsUpdater = (viewTag: number, props: unknown) => {
+          runOnJS(updater)(viewTag, props);
+        };
+      })();
+      JSPropUpdaterFabric.isInitialized = true;
+    }
+  }
+
+  public addOnJSPropsChangeListener(
+    animatedComponent: React.Component<unknown, unknown>
+  ) {
+    const viewTag = findNodeHandle(animatedComponent);
+    JSPropUpdaterFabric._tagToComponentMapping.set(viewTag, animatedComponent);
+  }
+
+  public removeOnJSPropsChangeListener(
+    animatedComponent: React.Component<unknown, unknown>
+  ) {
+    const viewTag = findNodeHandle(animatedComponent);
```

Comment:
```suggestion
    const shadowNodeWrapper = findNodeHandle(animatedComponent);
```

Reply:
Are you sure that findNodeHandle returns shadowNodeWrapper? According to the source code it should return just a number - https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/ReactNative/RendererImplementation.js#L47-L49. Also in every other place in our code we use `findNodeHandle` to obtain just a view tag. I think that `findHostInstance_DEPRECATED` returns a shadow node wrapper, but correct me if I'm wrong.

Reply:
You're right, `findHostInstance_DEPRECATED` returns a ShadowNodeWrapper. I'm not sure what's the result of `findNodeHandle`, I would need to check it. Anyway, the question is, why do we need this number on Fabric? I thought the only way to identify a RN component in C++ when updating props is to pass a ShadowNodeWrapper 🤔

Reply:
In Fabric, the view tag is not required. However, I utilized the result of `findNodeHandle()` (which is a number) to store React JavaScript components in the JavaScript dictionary called `_tagToComponentMapping`.

```js
class JSPropsUpdaterFabric implements JSPropsUpdater {
  private static _tagToComponentMapping = new Map();
```